src/DataTransferObject/AssignmentDtoCollection.php

```diff
@@ -29,39 +29,57 @@
 class AssignmentDtoCollection implements Countable, IteratorAggregate
 {
     /** @var AssignmentDto[] */
-    private $collection;
+    private $assignments;
 
     public function __construct(AssignmentDto ...$assignments)
     {
-        $this->collection = $assignments;
+        $this->assignments = $assignments;
     }
 
     public function add(AssignmentDto $dto): self
     {
-        $this->collection[] = $dto;
+        $this->assignments[] = $dto;
 
         return $this;
     }
 
-    public function merge(AssignmentDtoCollection $assignments): self
+    public function clear(): self
     {
-        foreach ($assignments as $assignment) {
-            $this->add($assignment);
-        }
+        $this->assignments = [];
 
         return $this;
     }
 
+    /**
+     * @return string[]
+     */
+    public function getAllUsernames(): array
+    {
+        return array_unique(
+            array_map(
+                static function (AssignmentDto $assignmentDto): string {
+                    return $assignmentDto->getUsername();
+                },
+                $this->assignments
+            )
+        );
+    }
+
+    public function isEmpty(): bool
+    {
+        return count($this) === 0;
```

Comment:
Should not count $this->assignments?

Reply:
It doesn't matter, but invoking the method is maybe a little bit nicer indeed. Changed it.